src/type.cpp

```diff
@@ -1599,6 +1599,15 @@ StructType::StructType(const std::string &n, const llvm::SmallVector<const Type
             lStructTypeMap[mname]->setBody(elementTypes);
         }
     }
+    // Create a unique anonymous struct name if we have an anonymous struct (name == "")
+    // Ensuring struct is created with a name, prevents each use of original
+    // struct from having different names causing type match errors.
+    if (name == "") {
+        char buf[16];
+        snprintf(buf, sizeof(buf), "$anon%d", count);
```

Comment:
We will create two names per anonymous struct - here and in the code above. Is intended?

Reply:
It'll not create two names. If it goes through the earlier path(i.e variability is mentioned, name won't be " "(empty) anymore. If unbounded, name will be set here..otherwise at the earlier place.

Reply:
Ok, makes sense.